libraries/AP_HAL_ChibiOS/hwdef/CSKY405/Readme.md

```diff
@@ -0,0 +1,100 @@
+# CSKY405 Flight Controller
+
+The CSKY405 is a flight controller produced by [ClearSky](http://csky.space/).
+
+## Features
+    Processor
+        STM32F405 168Mhz, 1MB 32-bit processor
+        AT7456E OSD
+    Sensors
+        BMI088 Acc/Gyro
+        BMP390 barometer
+    Power
+        2S - 6S Lipo input voltage with voltage monitoring
+        90A Cont., 215A peak current monitor
+        12V, 2A BEC for powering Video Transmitter controlled by GPIO
+        5V, 2.5A BEC for internal and peripherals
+    Interfaces
+        10x PWM outputs DShot capable
+        1x RC input
+        6x UARTs/serial for GPS and other peripherals
+        I2C port for external compass, airspeed, etc.
+        microSDCard for logging, etc.
+        USB-C port
+  
+
+## Overview
+
+![CSKY405](CSKY405.png)
+
+## Wiring Diagram
+
+![CSKY405 Wiring](CSKY405_wiring.png)
+
+## UART Mapping
+
+The UARTs are marked Rn and Tn in the above pinouts. The Rn pin is the
+receive pin for UARTn. The Tn pin is the transmit pin for UARTn.
+
+ - SERIAL0 -> USB
+ - SERIAL1 -> UART4  (MAVLink2 telem)
+ - SERIAL2 -> USART1 (Serial RC input) (DMA capable)
+ - SERIAL3 -> UART5  (GPS)
+ - SERIAL4 -> USART3 (User) (DMA capable)
+ - SERIAL5 -> USART6 (User) (TX DMA capable)
+ - SERIAL6 -> USART2 (RX tied to inverted SBUS RC input, but can be used as normal UART if :ref:`BRD_ALT_CONFIG<>` =1) 
+
+
+## RC Input
+
+RC input is configured on the SBUS pin (inverted and sent to UART2_RX). It supports all RC
+protocols except serial protocols such as CRSF, ELRS, etc. Those devices can be connected to USART1 TX and RX, instead.
+Fport can be connected to USART1 TX also, but will require an external bi-directional inverter and the ref:`SERIAL1_OPTION<SERIAL1_OPTION>' = 4 (HalfDuplex) set.
+   
+## OSD Support
+
+The CSKY405 supports using its internal OSD using OSD_TYPE 1 (MAX7456 driver). External OSD support such as DJI or DisplayPort is supported using USART6 or any other free USART6. See :ref:`common-msp-osd-overview-4.2` for more info.
+
+## PWM Output
+
+The CSKY405 supports up to 10 PWM outputs. S1-S4 outputs support DShot.
```

Comment:
you have defined pwm1-4as bidir, but do not advertize this, which is good, because they share dma, it might not work.  @andyp1per pls review and comment

Reply:
okay, I will remove it in this PR and add it in the future when I check if this PR is accepted